### PR TITLE
1210 js fix re route to client app

### PR DESCRIPTION
**Pull request by PlanetEfficacy**

### CODE REVIEW

File: app/controllers/application_controller.rb [hunk 1]
```diff
@@ -13,4 +14,25 @@ def configure_permitted_parameters
...
+
+    def check_authenticity
+      if authentic_client_id?
+        store_current_location
+      else
+        head :forbidden
+      end
+    end
+
+    def store_current_location
```
Comment (calaway) [inline app/controllers/application_controller.rb]: Consider changing this method name?

File: app/controllers/application_controller.rb [hunk 2]
```diff
@@ -1,6 +1,7 @@
 class ApplicationController < ActionController::Base
   protect_from_forgery with: :exception
   before_action :configure_permitted_parameters, if: :devise_controller?
+  before_action :check_authenticity, if: :request_from_client?
```
Comment (calaway) [inline app/controllers/application_controller.rb]: Does this need to be before every controller action?

File: app/controllers/users/sessions_controller.rb [hunk 3]
```diff
@@ -0,0 +1,32 @@
+class Users::SessionsController < Devise::SessionsController
+# before_action :configure_sign_in_params, only: [:create]
+
+  # GET /resource/sign_in
+  def new
```
Comment (calaway) [inline app/controllers/users/sessions_controller.rb]: delete please

File: app/controllers/application_controller.rb (hunk 1 above)
Comment (PlanetEfficacy) [inline app/controllers/application_controller.rb]: I'm going with `store_request_fullpath`. I'm also renaming `request_from_client?` to `client_request?`.

File: app/controllers/users/sessions_controller.rb (hunk 3 above)
Comment (PlanetEfficacy) [inline app/controllers/users/sessions_controller.rb]: Check.

File: config/initializers/doorkeeper.rb [hunk 4]
```diff
@@ -36,7 +41,7 @@
   # The controller Doorkeeper::ApplicationController inherits from.
   # Defaults to ActionController::Base.
   # https://github.com/doorkeeper-gem/doorkeeper#custom-base-controller
-  # base_controller 'ApplicationController'
+  base_controller 'ApplicationController'
```
Comment (PlanetEfficacy) [inline config/initializers/doorkeeper.rb]: Changed to DoorkeeperController

**Comment by bcgoss** [comment]
I'm so glad we solved this. This is much better than my idea.

**Pull request merged**